downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,422 @@
+
+  ```ballerina
+  public function main() {
+    int resultInt;
+    transaction {
+        resultInt = check calculateDefaultValue(true);
+        check commit;
+    } on fail {
+        io:println("Failed to initialize resultInt");
+    }
+    resultInt += 1; // Compilation error now.
+  }
+  ```
+
+- A bug that resulted incorrect type inference within query expressions when there is no expected type has been addressed. Previously, when iterating over a map without explicitly specifying an expected type, the resulting type of the query expression was erroneously inferred as an array. This misinterpretation has now been rectified and is properly restricted.
+  
+- ```ballerina
+  function filterEmployeesByDepartment(map<Employee> employees, string department) {
+    var result = from var e in employees // Compilation error now.
+        where e.department == department
+        select e.name;
+  }
+  ```
+
+- A bug that allowed ignoring possible completion with an error when using the `collect` clause in a query expression has been fixed.
+ 
+- ```ballerina
+  function calculateTotalSalary(stream<Employee, error?> strm, string dept) {
+    int total = from var {department, salary} in strm // Compilation error now.
+        where department == dept
+        collect sum(salary);
+  }
+  ```
+
+- A bug related to deciding the types of numeric literals has been fixed.
+
+    ```ballerina
+    2f|1 a = 2; // Compilation error now, `2` is considered to be `int`.
+    3d|6 b = 3; // Compilation error now, `3` is considered to be `int`.
+    3d|4f|6 b = 4; // Compilation error now, `4` is considered to be `int`.
+    ```
+
+- A bug that allowed an `anydata` value to be used in contexts expecting an `anydata & readonly` value has been fixed.
+
+    ```ballerina
+    public function main() {
+        anydata a = [];
+        anydata & readonly b = a; // Compilation error now.
+    }
+    ```
+  
+- A bug that allowed using object type inclusion with a `readonly` `service` object in a `readonly` non-`service` class has been fixed.
+
+    ```ballerina
+    public type A readonly & service object {
+        isolated remote function execute() returns int;
+    };
+  
+    readonly client class B {
+        *A; // Compilation error now.
+  
+        isolated remote function execute() returns int {
+            return 2;
+        }
+    }
+    ```
+
+- A bug that caused the runtime error messages to include the flattened representation of a union type instead of the name used in the type definition has been fixed.
+  
+    ```ballerina
+    type AC map<int>|string;
+    
+    public function main() returns error? {
+        map<anydata> x = {a: "Ballerina"};
+        AC|boolean _ = check x.cloneWithType();
+    }
+    ```
+  
+- A bug that resulted in an incorrect value when using a global variable, as a default value for the class field and function parameter has been fixed.
```

Comment:
```suggestion
- A bug that resulted in an incorrect value being used as the default value of a class field when the default value refers to a module-level variable that is also used in the default value of a function parameter has been fixed.
```

cc @chiranSachintha 

We need to use proper terms (module-level, not global).

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-dev-website/pull/7849